app/screens/more_channels/more_channels.js

```diff
@@ -143,15 +150,16 @@ export default class MoreChannels extends PureComponent {
     };
 
     doGetChannels = () => {
-        const {actions, currentTeamId} = this.props;
+        const {actions, currentTeamId, canShowArchivedChannels} = this.props;
         const {loading, term} = this.state;
 
         if (this.next && !loading && !term && this.mounted) {
             this.setState({loading: true}, () => {
-                actions.getChannels(
+                actions.loadPublicAndArchivedChannels(
                     currentTeamId,
                     this.page + 1,
-                    General.CHANNELS_CHUNK_SIZE
+                    General.CHANNELS_CHUNK_SIZE,
```

Comment:
for some reason I don't trust that the paging will work as intended here as it should when you switch between archived, public and all channels

Reply:
Is this comment still relevant if we're not doing archived, public, and all channels? Could you clarify why you don't trust it?

Reply:
That makes sense. I've added separate page counters for public and archived channels. Thanks!